clients/slo/slo.go

```diff
@@ -0,0 +1,213 @@
+	defer closeBody(resp)
+	if err != nil {
+		return api.Response{}, fmt.Errorf(errMsgWithId, "get", id, err)
+	}
+	return processResponse(resp, nil)
+}
+
+func (c Client) Create(ctx context.Context, body json.RawMessage) (api.Response, error) {
+	resp, err := c.client.Create(ctx, body, rest.RequestOptions{CustomShouldRetryFunc: rest.RetryIfTooManyRequests})
+	defer closeBody(resp)
+	if err != nil {
+		return api.Response{}, err // error message from the client is descriptive enough
+	}
+	return processResponse(resp, nil)
+}
+
+func (c Client) Update(ctx context.Context, id string, body json.RawMessage) (api.Response, error) {
+	getResp, err := c.Get(ctx, id)
+	if err != nil {
+		return api.Response{}, fmt.Errorf(errMsgWithId, "update", id, err)
+	}
+
+	version, err := getVersion(getResp)
+	if err != nil {
+		return api.Response{}, fmt.Errorf(errMsgWithId, "update", id, err)
+	}
+
+	resp, err := c.client.Update(ctx, id, version, body, rest.RequestOptions{CustomShouldRetryFunc: rest.RetryIfTooManyRequests})
+	defer closeBody(resp)
+	if err != nil {
+		return api.Response{}, fmt.Errorf(errMsgWithId, "update", id, err)
+	}
+	return processResponse(resp, nil)
+}
+
+func getVersion(resp api.Response) (string, error) {
+	var getStr struct {
+		Version string `json:"version"`
+	}
+
+	err := json.Unmarshal(resp.Data, &getStr)
+	if err != nil {
+		return "", err
+	}
+
+	return getStr.Version, nil
+}
+
+// Delete removes configuration for SLO with given ID from a server.
+func (c Client) Delete(ctx context.Context, id string) (api.Response, error) {
+	resp, err := c.client.Delete(ctx, id, rest.RequestOptions{CustomShouldRetryFunc: rest.RetryIfTooManyRequests})
+	closeBody(resp)
+	if err != nil {
+		return api.Response{}, fmt.Errorf(errMsgWithId, "delete", id, err)
+	}
+
+	if !rest.IsSuccess(resp) {
+		return api.Response{}, api.NewAPIErrorFromResponse(resp)
+	}
+	return api.NewResponseFromHTTPResponseAndBody(resp, nil), nil
+}
+
+func closeBody(httpResponse *http.Response) {
+	if httpResponse != nil && httpResponse.Body != nil {
+		_ = httpResponse.Body.Close()
+	}
+}
+
+func processResponse(httpResponse *http.Response, transform func([]byte) ([]byte, error)) (api.Response, error) {
+	var body []byte
+	var err error
+
+	if httpResponse.Body != nil {
+		if body, err = io.ReadAll(httpResponse.Body); err != nil {
+			return api.Response{}, api.NewAPIErrorFromResponseAndBody(httpResponse, body)
+		}
+	}
+
+	if !rest.IsSuccess(httpResponse) {
+		return api.Response{}, api.NewAPIErrorFromResponseAndBody(httpResponse, body)
+	}
+
+	if transform != nil {
+		if body, err = transform(body); err != nil {
+			return api.Response{}, api.NewAPIErrorFromResponseAndBody(httpResponse, body)
+		}
+	}
+
+	return api.NewResponseFromHTTPResponseAndBody(httpResponse, body), nil
+}
+
+type nextPage = string
+
+func processListResponse(httpResponse *http.Response, transform func(json.RawMessage) (nextPage string, data [][]byte, e error)) (nextPage, api.ListResponse, error) {
```

Comment:
You call this method with a fixed method, and it's only used once. Why pass it?

Reply:
changed